### DISCUSSION TITLE
queryContent is not defined error while calling from custom module using @nuxt/content v2

### DISCUSSION THREAD

**Discussion by jsoneaday** [nuxt/nuxt#18636] [Questions]
I am using defineNuxtModule and inside the setup function I define a build:done hook and call queryContent from it. Upon compilation I get error that queryContent is not defined. QueryContent works without issue in my other code. How can I import this function?

```
nuxt.hook('build:done', async () => {  
    const docs = queryContent('/').find();
```

**Comment by Baroshem** [comment]
From what I understood, queryContent is a composable from nuxt-content and composables are primarly meant to be called in the vue components in setup function, not in modules in the build hooks. 

Could you explain what are you trying to do here in this build hook?

**Comment by jsoneaday** [comment]
Yeah that's what it looked like but I was hoping there was some way to get it. As for what I'm doing I am trying to hook into the 'build:done' hook to run the algolia search indexer submit. I have a document app with markup files that I need indexed by algolia. The code, which is not working, was initially written for an older version of nuxt/content and so used $content to get at the md file content, but when I try and use that it errors. So then I tried switching to queryContent, as we use nuxt/content v2, but that is undefined. Is there no other way to get at the structured md content? ($content comes from const { @content } = require('./node_modules/nuxt/content')).

`
    nuxt.hook('build:done', async () => {     
            for (let i = 0; i < options.paths.length; i++) {
                const path = options.paths[i];
                const indexName = path.index || path.name;
                const deep = path.deep;
                console.log("path", path);
                // let docs = await $content(path.name, { deep }).fetch();
                const docs = queryContent('/').find();
                console.log("docs", docs);
                
                docs = docs.map((doc) => {
                    const newDoc = {}
                    path.fields.forEach((field) => (newDoc[field] = doc[field]))
                    newDoc.objectID = doc.slug
                    return newDoc
                })
                
                const client = algoliasearch(nuxt.options.runtimeConfig.appId, nuxt.options.runtimeConfig.apiKey)
                const index = client.initIndex(indexName)
        
                // clear the index in case any documents were removed
                await index.clearObjects()
                
                const { objectIDs } = await index.saveObjects(docs)
                console.log(
                    `Indexed ${objectIDs.length} records in Algolia for: ${indexName}`
                );
            }
        });
`

**Comment by jsoneaday** [comment]
I tried doing this import { queryContent } from '@nuxt/content/dist/runtime/composables/query'; but get error 

Package subpath './dist/runtime/composables/query' is not defined by "exports" in /Users/davidchoi/Work/philabs/archway-docs/node_modules/@nuxt/content/package.json

**Comment by jsoneaday** [comment]
I just tried getting the older $content via const { $content } = await import(`${nuxt.options.srcDir}/node_modules/@nuxt/content`);

But I get error illegal operation on a directory, read

**Comment by adamalfredsson** [comment]
I've had a similar issue, if it's any help, my approach was to prerender a json file with algolia data by adding `~/server/routes/algolia.json.ts`:

```typescript
import { serverQueryContent } from '#content/server'

export default defineEventHandler(async (event) => {
  const docs = await serverQueryContent(event).find()
  return docs.map((doc) => {
    return {
      objectID: doc._id,
      // add record data here
    }
  })
})
```

Make sure to prerender this route by the following configuration in `nuxt.config.ts`:

```typescript
export default defineNuxtConfig({
  nitro: {
    prerender: {
      routes: ['/algolia.json'],
    },
  },
}
```

Finally, after a deploy was succeeded for example, see if you can configure a webhook or something to fetch the json data and publish to algolia.

Hope it's to any help.
